tests/e2e/util.go

```diff
@@ -6100,41 +6108,78 @@ func CheckDevice(client clientset.Interface, sts *appsv1.StatefulSet, devicePath
+	startSizeInMB, dataSizeInMB int64) {
 	//Write some data to file first and then to raw block device
-	writeDataOnRawBlockVolume(ns, podName, devicePath, testData)
+	writeDataOnRawBlockVolume(f, ns, podName, devicePath, testdataFile, startSizeInMB, dataSizeInMB)
 	// Read the data to verify that is it same as what written
-	readDataFromRawBlockVolume(ns, podName, devicePath, testData)
+	readDataFromRawBlockVolume(f, ns, podName, devicePath, testdataFile, startSizeInMB, dataSizeInMB)
 }
 
 // writeDataOnRawBlockVolume writes test data to raw block device
-func writeDataOnRawBlockVolume(ns string, podName string, devicePath string, testData string) {
+func writeDataOnRawBlockVolume(f *framework.Framework, ns string, podName string, devicePath string, testdataFile string,
+	startSizeInMB, dataSizeInMB int64) {
 	cmd := []string{"exec", podName, "--namespace=" + ns, "--", "/bin/sh", "-c",
 		fmt.Sprintf("/bin/ls %v", devicePath)}
 	_, err := framework.RunKubectl(ns, cmd...)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-	cmd = []string{"exec", podName, "--namespace=" + ns, "--", "/bin/sh", "-c",
-		fmt.Sprintf("/bin/echo -ne '%v' > /tmp/data_to_write", testData)}
-	_, err = framework.RunKubectl(ns, cmd...)
+	_ = framework.RunKubectlOrDie(ns, "cp", testdataFile, fmt.Sprintf(
+		"%v/%v:/tmp/data_to_write", ns, podName))
 	framework.ExpectNoError(err, fmt.Sprintf("failed to write testdata inside the pod: %q", podName))
 
-	cmd = []string{"exec", podName, "--namespace=" + ns, "--", "/bin/sh", "-c",
-		fmt.Sprintf("/bin/dd if=/tmp/data_to_write of=%v", devicePath)}
-	_, err = framework.RunKubectl(ns, cmd...)
-	framework.ExpectNoError(err, fmt.Sprintf("failed to write device: %q inside the pod: %q", devicePath, podName))
+	// If startSizeInMB is given, fill 1M with testData from offset=startSizeInMB given, upto dataSizeInMB.
+	// Otherwise write the testData given from offset=0.
+	gomega.Expect(dataSizeInMB).NotTo(gomega.BeZero())
+	for i := int64(0); i < dataSizeInMB; i = i + 1 {
+		seek := fmt.Sprintf("%v", startSizeInMB+i)
+		cmd = []string{"exec", podName, "--namespace=" + ns, "--", "/bin/sh", "-c",
+			fmt.Sprintf("/bin/dd if=/tmp/data_to_write of=%v bs=1M count=1 conv=fsync seek=%v",
+				devicePath, seek)}
+		_, err = framework.RunKubectl(ns, cmd...)
+		framework.ExpectNoError(err, fmt.Sprintf("failed to write device: %q inside the pod: %q", devicePath, podName))
+	}
+	cmd = []string{"--namespace=" + ns, "exec", podName, "--", "/bin/sh", "-c", "rm /tmp/data_to_write"}
+	_ = framework.RunKubectlOrDie(ns, cmd...)
 }
 
 // readDataFromRawBlockVolume reads data from raw block device and verifies it against given input
-func readDataFromRawBlockVolume(ns string, podName string, devicePath string, testData string) {
+func readDataFromRawBlockVolume(f *framework.Framework, ns string, podName string, devicePath string, testdataFile string,
```

Comment:
since we are verifying the data as well, can you change the func name to indicate the same

Reply:
done